pkg/cmd/roachtest/tests/online_restore.go

```diff
@@ -242,6 +153,79 @@ func registerOnlineRestore(r registry.Registry) {
 	}
 }
 
+func registerOnlineRestoreCorrectness(r registry.Registry) {
+	sp := restoreSpecs{
+		hardware: makeHardwareSpecs(hardwareSpecs{workloadNode: true}),
+		backup: makeRestoringBackupSpecs(backupSpecs{
+			nonRevisionHistory: true,
+			version:            fixtureFromMasterVersion,
+			workload:           tpccRestore{opts: tpccRestoreOptions{warehouses: 10, workers: 1, waitFraction: 0}}}),
+		timeout:                15 * time.Minute,
+		suites:                 registry.Suites(registry.Nightly),
+		restoreUptoIncremental: 1,
+		namePrefix:             "online/correctness",
+	}
+	sp.initTestName()
+	r.Add(
+		registry.TestSpec{
+			Name:                sp.testName,
+			Owner:               registry.OwnerDisasterRecovery,
+			Cluster:             sp.hardware.makeClusterSpecs(r, sp.backup.cloud),
+			Timeout:             sp.timeout,
+			CompatibleClouds:    sp.backup.CompatibleClouds(),
+			Suites:              sp.suites,
+			SkipPostValidations: registry.PostValidationReplicaDivergence,
+			Run: func(ctx context.Context, t test.Test, c cluster.Cluster) {
+				defaultSeed := crdbworkload.NewUint64RandomSeed().Seed()
+				defaultFakeTime := uint32(timeutil.Now().Unix())
+				regRestoreSpecs, regWorkload := initCorrectnessRestoreSpecs(
+					t, sp, defaultSeed, defaultFakeTime, "-reg.trace",
+				)
+				onlineRestoreSpecs, onlineWorkload := initCorrectnessRestoreSpecs(
+					t, sp, defaultSeed, defaultFakeTime, "-online.trace",
+				)
+
+				runRestore(ctx, t, c, regRestoreSpecs, false, true, false)
+				details, err := c.RunWithDetails(
+					ctx,
+					t.L(),
+					option.WithNodes([]int{regRestoreSpecs.hardware.getWorkloadNode()}),
+					fmt.Sprintf("cat %s", regWorkload.opts.queryTraceFile),
+				)
+				require.NoError(t, err, "failed to retrieve query trace for regular restore")
+				regQueryTrace := details[0].Stdout
```

Comment:
Just a note about the memory requirements for storing the output for comparison. On a 15-minute timeout, this created a query trace of about ~15 MB.